[MSE in Worker] ASSERT(canSafelyBeUsed());

jyavenard opened the pull request:
#### dd8b32e43a1aca81e73e37b9b0378966616d27bc
<pre>
[MSE in Worker] ASSERT(canSafelyBeUsed());
<a href="https://bugs.webkit.org/show_bug.cgi?id=270894">https://bugs.webkit.org/show_bug.cgi?id=270894</a>
<a href="https://rdar.apple.com/124500515">rdar://124500515</a>

Reviewed by Youenn Fablet.

Only ever access the MediaSource::m_mediaElement member on the main thread.

Covered by existing tests.

* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::detachFromElement):
(WebCore::MediaSource::attachToElement):
(WebCore::MediaSource::stop):
* Source/WebCore/Modules/mediasource/MediaSource.h: We can&apos;t use check the value of m_mediaElement
to determine if one is attached or not. Add boolean instead.

Canonical link: <a href="https://commits.webkit.org/276100@main">https://commits.webkit.org/276100@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d1daff42472b6d02cec6cfc96097bb8091670834

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43735 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22784 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46174 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46380 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39838 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46039 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26648 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20191 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36118 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44309 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19814 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37676 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17103 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/17349 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38746 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1792 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/39906 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39051 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/47936 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18765 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/15344 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/42924 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20172 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41609 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9732 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20364 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/19798 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf reviewed (approved):
the m_isAttached change is probably changing behaviour.
In case the media element would be GCed, m_mediaElement would be nullptr and it seems we could call attachToElement again on the same MediaSource without having called detachFromElement.

Can we add a test for this if possible?

jyavenard committed:
[MSE in Worker] ASSERT(canSafelyBeUsed());
https://bugs.webkit.org/show_bug.cgi?id=270894
rdar://124500515

Reviewed by Youenn Fablet.

Only ever access the MediaSource::m_mediaElement member on the main thread.

Covered by existing tests.

* Source/WebCore/Modules/mediasource/MediaSource.cpp:
(WebCore::MediaSource::detachFromElement):
(WebCore::MediaSource::attachToElement):
(WebCore::MediaSource::stop):
* Source/WebCore/Modules/mediasource/MediaSource.h: We can't use check the value of m_mediaElement
to determine if one is attached or not. Add boolean instead.

Canonical link: https://commits.webkit.org/276100@main
